pypdf/_writer.py

```diff
@@ -981,6 +1005,75 @@ def reattach_fields(
                 lst.append(ano)
         return lst
 
+    def get_pages_showing_field(
```

Comment:
Do we have the chance to move this to a common utility method to avoid the duplicates?

Reply:
Yes:  I plan to propose a PR that would create a common `PdfDocument` that would mutualize as much as possible code. I did such work a longtime ago on PyPDF4.
This should come in the next weeks